### PR TITLE
Failing when no match found instead of warning

### PR DESCRIPTION
**Pull request by ds-ms**

### CODE REVIEW

File: Tasks/DotNetCoreCLI/Common/utility.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import * as tl from "vsts-task-lib/task";
 import * as nuGetGetter from "nuget-task-common/NuGetToolGetter";
 
-export function getProjectFiles(projectPattern: string[]): string[] {
-    if (projectPattern.length == 0) {
-        return [""];
```
Comment (sachinma) [inline Tasks/DotNetCoreCLI/Common/utility.ts]: Lets keep the original version

File: Tasks/DotNetCoreCLI/pushcommand.ts [hunk 2]
```diff
@@ -1,39 +1,34 @@
...
     let buildIdentityDisplayName: string = null;
     let buildIdentityAccount: string = null;
     try {
         // Get list of files to publish
         const searchPattern = tl.getPathInput("searchPatternPush", true, false);
-
-        let findOptions: tl.FindOptions = <tl.FindOptions>{};
-        let matchOptions: tl.MatchOptions = <tl.MatchOptions>{};
-        const filesList = tl.findMatch(undefined, searchPattern, findOptions, matchOptions);
+        const filesList = utility.searchFiles(searchPattern);
```
Comment (sachinma) [inline Tasks/DotNetCoreCLI/pushcommand.ts]: revert this

File: Tasks/DotNetCoreCLI/restorecommand.ts [hunk 3]
```diff
@@ -20,8 +20,11 @@ export async function run(): Promise<void> {
         const projectSearch = tl.getDelimitedInput("projects", "\n", false);
 
         // if no projectSearch strings are given, use "" to operate on the current directory
-        const projectFiles = utility.getProjectFiles(projectSearch);
-
+        const projectFiles = projectSearch.length != 0 ? utility.searchFiles(projectSearch) : [""];
```
Comment (sachinma) [inline Tasks/DotNetCoreCLI/restorecommand.ts]: revert this

File: Tasks/DotNetCoreCLI/Tests/L0.ts [hunk 4]
```diff
@@ -180,98 +241,168 @@ describe('DotNetCoreExe Suite', function () {
...
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        tr.run();
+
+        // TODO: Add Zip
+        assert(tr.invokedToolCount == 1, 'should have invoked tool');
+        assert(tr.succeeded, 'task should have succeeded');
+        done();
+    });
+
+    it('publish works with no project file is specified', (done: MochaDone) => {
```
Comment (sachinma) [inline Tasks/DotNetCoreCLI/Tests/L0.ts]: This look like a merge issue. This is coming twice.

File: Tasks/DotNetCoreCLI/pushcommand.ts [hunk 5]
```diff
@@ -31,9 +29,8 @@ export async function run(): Promise<void> {
             }
         });
 
-        if (filesList && filesList.length < 1)
-        {
-            tl.setResult(tl.TaskResult.Succeeded, tl.loc("Info_NoPackagesMatchedTheSearchPattern"));
+        if (filesList.length < 1) {
```
Comment (sachinma) [inline Tasks/DotNetCoreCLI/pushcommand.ts]: can fileList be null

Comment (ds-ms) [inline Tasks/DotNetCoreCLI/pushcommand.ts]: If it is null, it'll fail in the forEach step itself. findMatch will never return null https://github.com/Microsoft/vsts-task-lib/blob/master/node/task.ts#L1525

File: Tasks/DotNetCoreCLI/Tests/L0.ts (hunk 4 above)
Comment (ds-ms) [inline Tasks/DotNetCoreCLI/Tests/L0.ts]: Yes, I'll fix it

**Comment by ds-ms** [comment]
@sachinma searchPattern in push and pack can never be empty (cause it's a required field), seems like a redundant addition.

**Pull request merged**